File: 102. Binary Tree Level Order Traversal/102. Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []

        res = []
        q = deque([root])

        while q:
            level_length = len(q)
            level = []
            for _ in range(level_length):
                n = q.popleft()
                level.append(n.val)

                if n.left:
                    q.append(n.left)
                if n.right:
                    q.append(n.right)

            res.append(level)

        return res

File: 102. Binary Tree Level Order Traversal/102. Binary Tree Level Order Traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
  if (!root) {
    return [];
  }

  const res = [];
  const q = [root];

  while (q.length > 0) {
    level_length = q.length;
    level = [];

    for (let i = 0; i < level_length; i++) {
      const n = q.shift();
      level.push(n.val);

      if (n.left) {
        q.push(n.left);
      }

      if (n.right) {
        q.push(n.right);
      }
    }

    res.push(level);
  }

  return res;
};
